packages/flame/test/effects/sequence_effect_test.dart

```diff
@@ -0,0 +1,281 @@
+        final game = FlameGame()..onGameResize(Vector2.all(1000));
+        await game.ensureAdd(component);
+        game.update(0);
+
+        final forwardPath = <Vector2>[
+          for (var i = 0; i < 10; i++) Vector2(i * 0.1, 0),
+          for (var i = 10; i > 0; i--) Vector2(i * 0.1, 0),
+          for (var i = 0; i < 10; i++) Vector2(0, i * 0.1),
+          for (var i = 10; i > 0; i--) Vector2(0, i * 0.1),
+        ];
+        final expectedPath = [
+          ...forwardPath,
+          Vector2.zero(),
+          ...forwardPath.reversed,
+        ];
+        for (final p in expectedPath) {
+          expect(component.position, closeToVector(p.x, p.y, epsilon: 1e-14));
+          game.update(0.1);
+        }
+        game.update(0.001);
+        expect(component.position, closeToVector(0, 0));
+        expect(effect.controller.completed, true);
+      });
+
+      test('sequence in sequence', () async {
+        EffectController duration(double t) => EffectController(duration: t);
+        final effect = SequenceEffect(
+          [
+            MoveEffect.by(Vector2(10, 10), duration(1)),
+            SequenceEffect(
+              [
+                MoveEffect.to(Vector2(20, 0), duration(1)),
+                MoveEffect.to(Vector2(30, 10), duration(1)),
+              ],
+              alternate: true,
+              repeatCount: 2,
+            ),
+            MoveEffect.by(Vector2(0, 20), duration(2)),
+            SequenceEffect(
+              [
+                MoveEffect.by(Vector2(1, 0), duration(1)),
+                MoveEffect.by(Vector2(0, 1), duration(1)),
+              ],
+              repeatCount: 5,
+            ),
+          ],
+          alternate: true,
+        );
+        expect(effect.controller.duration, 42);
+
+        final component = PositionComponent()..add(effect);
+        final game = FlameGame()..onGameResize(Vector2.all(1000));
+        await game.ensureAdd(component);
+        game.update(0);
+
+        final forwardPath = <Vector2>[
+          for (var i = 0; i < 100; i++) Vector2(i * 0.1, i * 0.1),
+          for (var i = 0; i < 100; i++) Vector2(10 + i * 0.1, 10 - i * 0.1),
+          for (var i = 0; i < 100; i++) Vector2(20 + i * 0.1, i * 0.1),
+          for (var i = 100; i > 0; i--) Vector2(20 + i * 0.1, i * 0.1),
+          for (var i = 100; i > 0; i--) Vector2(10 + i * 0.1, 10 - i * 0.1),
+          for (var i = 0; i < 100; i++) Vector2(10 + i * 0.1, 10 - i * 0.1),
+          for (var i = 0; i < 100; i++) Vector2(20 + i * 0.1, i * 0.1),
+          for (var i = 100; i > 0; i--) Vector2(20 + i * 0.1, i * 0.1),
+          for (var i = 100; i > 0; i--) Vector2(10 + i * 0.1, 10 - i * 0.1),
+          for (var i = 0; i < 200; i++) Vector2(10, 10 + i * 0.1),
+          for (var j = 0; j < 5; j++)
+            for (var i = 0; i < 200; i++)
+              Vector2(i < 100 ? i * 0.01 : 1, i < 100 ? 0 : i * 0.01 - 1)
+                ..add(Vector2(10 + j * 1.0, 30 + j * 1.0)),
+        ];
```

Comment:
These are a lot of magic numbers for someone not familiar with the code, Maybe some of the numbers could be broken out to named variables that makes it easier to read?

Reply:
And maybe a comment explaining it too.

Reply:
Ok, i've turned the magic numbers into constants